python/eos/figure/item.py

```diff
@@ -1433,6 +1433,101 @@ def draw(self, ax):
+    :type resolution: int
+    :param variables: The names of the variables to be used as x and y coordinates.
+    :type variables: tuple[str, str]
+    """
+    fixed_kinematics:dict[str,float]|None=None
+    fixed_parameters:dict[eos.QualifiedName,float]|None=None
+    fixed_parameters_from_file:str|None=None
+    observable:eos.QualifiedName
+    options:dict[str, str]|None=None
+    variables:tuple[str,str]
+    resolution:int=100
+
+    def __post_init__(self):
+        eos.info(f'Handling item to plot {self.observable} in the complex plane')
+        self._observable_entry = eos.Observables()[self.observable]
+        valid_kinematic_variables = {kv for kv in self._observable_entry.kinematic_variables()}
+
+        # Create kinematics
+        self._kinematics = eos.Kinematics()
+        if self.fixed_kinematics:
+            for k, v in self.fixed_kinematics.items():
+                if k not in valid_kinematic_variables:
+                    raise ValueError(f"Kinematic variable '{k}' does not match any of the declared kinematic variables '{self.observable}': {valid_kinematic_variables.__repr__()}")
+                self._kinematics.declare(k, v)
+
+        # Create parameters
+        self._parameters = eos.Parameters.Defaults()
+        if self.fixed_parameters_from_file:
+            eos.warn('Overriding parameters from file')
+            self._parameters.override_from_file(self.fixed_parameters_from_file)
+
+        if self.fixed_parameters:
+            if self.fixed_parameters_from_file:
+                eos.warn('Overriding values read from \'parameters-from-file\' with explicit values in \'parameters\'')
+
+            for key, value in self.fixed_parameters.items():
+                self._parameters.set(key, value)
+
+        # Declare variable that is either parameter or kinematic
+        self._variable = None
+
+        # Do the variables correspond to one of the kinematic variables?
+        if self.variables[0] in valid_kinematic_variables:
+            self._kvx = self._kinematics.declare(self.variables[0], _np.nan)
+        else:
+            raise ValueError(f"Value of the first element of 'variables' ('{self.variables[0]}') is not the name of a kinematic variable for observable '{self.observable}'")
+
+        if self.variables[1] in valid_kinematic_variables:
+            self._kvy = self._kinematics.declare(self.variables[1], _np.nan)
+        else:
+            raise ValueError(f"Value of the second element of 'variables' ('{self.variables[1]}') is not the name of a kinematic variable for observable '{self.observable}'")
+
+        # Create options
+        self._options = eos.Options()
+        if self.options:
+            for key, value in self.options.items():
+                self._options.declare(key, value)
+
+        self._observable = eos.Observable.make(self.observable, self._parameters, self._kinematics, self._options)
+
+        self._xvalues = _np.linspace(-2.0, +2.0, self.resolution)
```

Comment:
This should be the `range` provided by the user, no?

Reply:
Fixed the range. The shape can be nicely done as part of the transformation inside the observable.